[pzl111] iP

pzl111 opened the pull request:
## Some PR request for iP

* Why is
* CS2103T
* Taking up
* So much time

1. uno
2. dos
3. tres
4. cuatro

```java
public class EndClass {
        public static void main(String[] args) {
                System.out.print("I want to sleep");
        }
}
```


- [X] Week 1 complete 👍 
- [X] Week 2 complete 🎉 
- [X] Week 3 complete 🎉 
- [ ] Week 4 incomplete 😢 

> Very cool and amazing blockquote

Link to awesome website aka my orbital page [GoWhereDoWhat](https://gwdw-orbital.web.app/)

This method `System.out.println()` prints out my desire to sleep.

We have heavy text formatting here, inclusive of **bolded words**, *italic words*, words that have been ~~striked off~~, with <sub>subscript</sub> text and <sup>superscript</sup> text. Also the color white `#ffffff` is nice.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

pzl111 committed:
Completed Level-0

pzl111 committed:
Completed Level-1

pzl111 committed:
Completed Level-2

pzl111 committed:
Completed Level-3

pzl111 committed:
Completed Level-4

pzl111 committed:
Level-6 completed

pzl111 committed:
Completed Level-5

pzl111 committed:
Completed A-TextUiTesting

pzl111 committed:
Fixed A-TextUiTesting not working because of 'FC' command not found

pzl111 committed:
Complete Level-7

pzl111 committed:
Add LocalDate imports

pzl111 committed:
Finish implementing dateTime

pzl111 committed:
Add error checking for dateTime

pzl111 committed:
Complete Level-8

pzl111 committed:
Change data.txt

pzl111 committed:
Merge branch 'branch-Level-8'

pzl111 committed:
Complete A-MoreOOP

pzl111 committed:
Make Command Interface public

pzl111 committed:
Complete A-Packages

pzl111 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-Increments

pzl111 committed:
Complete A-Gradle

pzl111 committed:
Setup JUnit

pzl111 committed:
Testing Todo with JUnit

pzl111 committed:
Complete A-JUnit

pzl111 committed:
Complete A-Jar

pzl111 committed:
Merge branch 'branch-Increments'

pzl111 committed:
Complete Level-9

pzl111 committed:
Complete A-JavaDoc

pzl111 committed:
Configure A-CodingStandard

pzl111 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/smolbrain/Parser.java
#	src/main/java/smolbrain/task/Task.java

pzl111 committed:
Merge branch 'branch-A-CodingStandard'

pzl111 committed:
Fix coding standard

pzl111 committed:
Fix javadocs

pzl111 committed:
Initialise JavaFX with template files

tayruxin commented on src/main/java/smolbrain/task/TaskList.java:
```diff
@@ -0,0 +1,125 @@
+        if (tasklist.size() > 0) {
+            for (int i = 0; i < tasklist.size(); i++) {
+                if (tasklist.get(i).contain(keyword)) {
+                    ui.showMessage((i + 1) + ". " + tasklist.get(i));
```
Perhaps it is better to reduce the nesting of the happy path inside a loop by using a continue statement.

![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/106758300/2d5b044a-b93f-499a-8d6e-bec588d0e20d)

tayruxin commented on src/test/java/smolbrain/ParserTest.java:
```diff
@@ -0,0 +1,276 @@
+public class ParserTest {
+
+    @Test
+    public void parseTodoTest1() {
```
Will it be better to explicitly state what this test code is testing instead of labelling it as test1 and test2?

tayruxin commented on src/test/java/smolbrain/ParserTest.java:
```diff
@@ -0,0 +1,276 @@
+package smolbrain;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+
+import smolbrain.exception.InvalidDateTimeException;
+import smolbrain.exception.InvalidNumberException;
+import smolbrain.exception.InvalidRangeException;
+import smolbrain.exception.MissingDescriptionException;
+import smolbrain.exception.MissingTimeException;
+import smolbrain.task.Task;
```
Good use of spacing to separate the various import statements

tayruxin commented on src/main/java/smolbrain/Storage.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    /**
+     * Writes the text passed into the save file, overwriting any previous data.
+     *
+     * @param text Text to write to save file.
+     * @throws IOException If there is problems accessing the save file.
```
Good job! The javadoc was written really well

tayruxin reviewed (commented):
Overall, a very good job! The JavaDoc were written well and the code is well organised with several packages.

pzl111 committed:
Done integrating smolbrain parsing, with text overflow

pzl111 committed:
Shrink height of dialog box

pzl111 committed:
Shift profile picture smaller and in corner

pzl111 committed:
complete dialog box, with text bg color, paddings, set window title

pzl111 committed:
create another userdialog for user input

pzl111 committed:
Complete UI design

pzl111 committed:
Complete Level-10

pzl111 committed:
Merge branch 'branch-Level-10'

Jonyxzx commented on src/main/java/smolbrain/command/AddCommand.java:
```diff
@@ -0,0 +1,48 @@
+public class AddCommand implements Command {
+
+    private Task task;
+    private boolean loading;
```
the boolean would be great to start with is/has.

Jonyxzx commented on src/main/java/smolbrain/Parser.java:
```diff
@@ -0,0 +1,332 @@
+
+        String[] words = input.split(" ");
+
+        switch (words[0]) {
```
Great indentation of switch case operation

Jonyxzx commented on src/main/java/smolbrain/Smolbrain.java:
```diff
@@ -0,0 +1,57 @@
+    private static boolean loading = true;
+
+    private Storage storage;
+    private TaskList tasks;
```
Good use of plural form for a collection

Jonyxzx commented on src/main/java/smolbrain/task/Event.java:
```diff
@@ -0,0 +1,52 @@
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+        return "[E]" + super.toString() + " (from: " + this.from.format(formatter)
+                + " to: " + this.to.format(formatter) + ")";
```
Good use of spacing to prevent long lines

Jonyxzx commented on src/main/java/smolbrain/task/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+ */
+public class TaskList {
+
+    private ArrayList<Task> tasklist;
```
A collection should be in plural form

Jonyxzx reviewed (commented):
Overall coding style is great, many booleans do not have is/has which is just a minor oversight.

pzl111 committed:
Change run configuration for gradle

pzl111 committed:
Complete code qualtiy, by implementing more happy paths, renaming booleans to starting with "is"

pzl111 committed:
Complete A-Assertions

pzl111 committed:
Merge pull request #1 from pzl111/branch-A-Assertions

Complete A-Assertions

pzl111 committed:
Merge branch 'master' of https://github.com/pzl111/ip

pzl111 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	data.txt

pzl111 committed:
Merge pull request #2 from pzl111/branch-A-CodeQuality

Branch a code quality

pzl111 committed:
Merge branch 'master' of https://github.com/pzl111/ip

pzl111 committed:
Add priority command

Different tasks have different priorities that signify its urgency, but it cannot be included currently.

Tasks can now have a priority level assigned to them from 0 to 3. The default priority level of a task is set to 0.

To change the priority level of a task, use the priority command "priority id level",
where id is the index of the task and level is the priority level to set to.

pzl111 committed:
Merge branch 'branch-BCD-Extension'

pzl111 committed:
Add welcome message for new users

Currently there is no difference in the greeting message for returning or new users.
Hence it is difficult for a user to know if he or she has used the chatbot before.

With the addition of a welcome message of "Welcome new user!!!", the user
will definitely know if this is a new account with new data.

It can also be used to verify that all previous data has been cleared and
the chatbot is reset.

pzl111 committed:
Change the chatbot message when there are no tasks in chatbot.

Previously, regardless of how many tasks are in the chatbot, when the user
sends the list command, it gives the same message "Here are the tasks in your list:",
followed by the list of tasks. However, when there are no tasks, the message seems to be
strange where it describes an empty list.

The message is now changed to show "You have no tasks currently." when there are no
tasks in the chatbot. This is now more natural as a dialog and is straight and concise.

pzl111 committed:
Screenshot of Ui uploaded to docs folder

pzl111 committed:
Update README.md

pzl111 committed:
Uploaded smaller screenshot

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Add more testing for Todo method

pzl111 committed:
Update README.md

pzl111 committed:
Fix coding standards

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md

pzl111 committed:
Edit code standards

pzl111 committed:
Update README.md

pzl111 committed:
Update README.md
